tools/dockerfile-llb-frontend/test/buildkit/buildkit_test.go

```diff
@@ -0,0 +1,51 @@
+package buildkit
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+
+	e2eutil "kraftkit.sh/tools/dockerfile-llb-frontend/test"
+)
+
+type BuildKitDriver struct {
+	e2eutil.CmdExec
+}
+
+func (b BuildKitDriver) Run(localRegistry string, pluginImage string, appDir string, t *testing.T) ([]byte, error) {
+	if err := b.buildkitWorkersAvailable(); err != nil {
+		t.Fatalf("buildkit daemon not ready for testing: %v", err)
+	}
+	// Define frontend image name
+	frontendImage := fmt.Sprintf("%s/%s", localRegistry, pluginImage)
+
+	out, err := b.Exec("buildctl", "build", "--frontend", "gateway.v0", "--opt", "source="+frontendImage, "--local", "context=../apps/"+appDir)
```

Comment:
`buildctl` program will need to be installed and correctly configured in GitHub actions in order for this to be invoked:

https://github.com/unikraft/kraftkit/blob/staging/.github/workflows/gotests.yaml#L15

Reply:
Thanks for pointing that out. 

I've modified the test suite to omit the BuildKit end-to-end tests if the BuildKit daemon/buildctl is not available in the environment. The primary reason for this adjustment is the indirect invocation of BuildKit by the docker suite, which in my assessment, provides adequate coverage for our current needs. Running the BuildKit tests directly would be beneficial when debugging a standalone BuildKit instance. It's very useful for in-depth troubleshooting with a debugger but I'm not convinced of its added value in our CI environment.

Furthermore, integrating it into our CI would not only increase the test duration but would also complicate our pipeline. Setting up a rootless BuildKit daemon would necessitate a change in the test environment topology and a multi-step configuration which I believe might introduce potential fragility and maintenance overhead.

I've documented that in the README.md and the BuildKit test file.